### PR TITLE
Add information on Platforms to docs

### PR DESCRIPTION
**Pull request by wxtim**
#170

### CODE REVIEW

**Review by kinow — APPROVED** [review]
Few comments, no blockers. Rest looks good to me, rendered as HTML, looking really nice! Thanks @wxtim !

image

@oliver-sanders @wxtim I think later our Sphinx theme/style is gonna need an extra class for the HTML element for the caption.

image

**Comment by wxtim** [comment]
> @oliver-sanders @wxtim I think later our Sphinx theme/style is gonna need an extra class for the HTML element for the caption.

I noticed this, but I really wanted the caption. I think can finesse the style later.

**Comment by kinow** [comment]
>I noticed this, but I really wanted the caption. I think can finesse the style later.

:+1: doesn't need to be addressed here, can be done later IMO.

>Attempting to google this resulted in lots of bald cat memes.

:laughing: :cat:

**Comment by oliver-sanders** [comment]
> @oliver-sanders @wxtim I think later our Sphinx theme/style is gonna need an extra class for the HTML element for the caption.

https://github.com/cylc/cylc-doc/issues/210

**Review by datamel — APPROVED** [review]
Looks good. Minor comments.

File: src/7-to-8/major-changes/platforms.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+               batch system = slurm
+            [[[remote]]]
+               host = my_supercomputer
+
+   Read this section if your workflows jobs run on a remote computer or if
+   you see the following warning on running ``cylc validate``:
+
+   .. code-block::
+
+      WARNING - Task XXX: deprecated "host" and "batch system" will be removed at Cylc 9
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:28]:
```suggestion
   .. code-block:: console

      WARNING - Task <task>: deprecated "host" and "batch system" will be removed at Cylc 9
```

File: src/7-to-8/major-changes/platforms.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Overview
+--------
+
+Cylc 7 defines settings for remote :term:`jobs <job>` in each
+:term:`task's <task>` definition.
+
+Cylc 8 allows site administrators to configure :term:`platforms <platform>`.
+Each platform is a group of settings. The user needs only to
+select a platform for their tasks, and does not need to set all
+the settings.
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:40]:
```suggestion
Cylc 8 allows site administrators (and users) to configure
  :term:`platforms <platform>` in ``global.cylc``. A platform can have
  multiple hosts with associated platform-specific settings. Users only need to
  select the platform for their task jobs.
 ```

File: src/7-to-8/major-changes/platforms.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+.. warning::
+
+   Cylc 8 contains upgrade logic which should handle Cylc 7
+   settings in most cases. Cylc 8 will warn you when it runs
+   the upgrade logic. You should upgrade these parts of your
+   workflows. Cylc 9 will deprecate the upgrade logic.
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:47]:
```suggestion
   workflows. Deprecated Cylc 7 settings will be removed at Cylc 9.
```

File: src/7-to-8/major-changes/platforms.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+
+Example
+-------
+
+Here is an example Cylc 7 :term:`graph`:
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:53]: Not a graph!

```suggestion
Here are some example Cylc 7 task definitions:
```

File: src/7-to-8/major-changes/platforms.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            batch system = pbs
+
+      [[mytask_login_to_hpc_and_submit]]
+         [[[remote]]]
+            host = $(supercomputer_login_node_selector_script)
+         [[[job]]]
+            batch system = slurm
+
+
+Which will result in Cylc Running:
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:75]:
```suggestion
This will result in Cylc running:
```

File: src/7-to-8/summary.rst [hunk 6]
```diff
@@ -166,6 +167,13 @@ other hosts on the same platform to interact with task jobs.
           # Platforms can have the same hosts with different job runners.
           platform = hpc1_background
 
+
+.. warning::
+
+   Cylc 8 will attempt to pick a sensible platform for your Cylc 7 settings,
+   but this upgrade logic will be removed at Cylc 9.
```
Comment (hjoliver) [inline src/7-to-8/summary.rst:174]:
```suggestion
   These deprecated settings will be removed at Cylc 9.
``` 
(The upgrade logic gets removed, but perhaps more importantly the deprecated logic becomes illegal/obsolete.

File: src/7-to-8/major-changes/platforms.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      [[mytask_cylc_server]]
+
+      [[mytask_big_server]]
+         platform = linuxbox42
+
+      [[mytask_submit_local_to_remote_computer]]
+         platform = pbs_local
+
+      [[mytask_login_to_hpc_and_submit]]
+         platform = $(supercomputer_login_node_selector_script)
```
Comment (hjoliver) [inline src/7-to-8/major-changes/platforms.rst:98]: This section probably needs a more explicit pointer to platforms config documentation here, not just several inlined links on the word "platforms".  E.g. *See blah for full documentation of platforms configuration.*

**Review by hjoliver — CHANGES_REQUESTED** [review]
Just a few minor suggestions.

File: src/7-to-8/major-changes/platforms.rst (hunk 7 above)
Comment (wxtim) [reply src/7-to-8/major-changes/platforms.rst:137]: Have a look at my update.

**Review by hjoliver — APPROVED** [review]
All good, thanks @wxtim :+1:

**Pull request merged**